### DISCUSSION TITLE
Check existence of mainClass

### DISCUSSION THREAD

**Discussion by erikvanoosten** [sbt/sbt#6867] [Ideas]
After a refactoring the `mainClass` configured in `build.sbt` was no longer correct. I needed a full build/test/deploy cycle to find out.

Wouldn't it be nice if sbt (or the sbt packager) would check this and simply fail the build?

**Comment by eed3si9n** [comment]
I guess from the sbt's point of view if the build user overrode `mainClass` setting key, they know what they are doing.

Since sbt-native-packager already contains a bunch of [validators](https://github.com/sbt/sbt-native-packager/blob/df35f5f39b015892e57069f566da1aae160e5517/src/main/scala/com/typesafe/sbt/packager/universal/UniversalPlugin.scala#L131-L135), maybe you or someone else could add one for this. Relevant portion in sbt is probably https://github.com/sbt/sbt/blob/e0f309f4c6d68be824a92cfbfa4577a82056d23d/main/src/main/scala/sbt/Defaults.scala#L989-L1008

**Reply by erikvanoosten** [reply]
Thanks, that is a good pointer. Its not so hard to add a check there. The check would simply check if `mainClass.value` is in `discoveredMainClasses.value`. But now I am wondering: should we raise an error, or just log a warning?

WDYT?

**Reply by erikvanoosten** [reply]
To answer my own question: just logging a warning would not help a lot. I guess it should be an error unless you suppress it with an additional setting. I'll create a PR to propose something...

**Comment by erikvanoosten** [comment]
I created https://github.com/sbt/sbt/pull/6876 to address this. Any feedback is appreciated.
